Sources/SwiftSyntaxBuilder/Buildables.swift.gyb

```diff
@@ -0,0 +1,172 @@
+%{
+  from gyb_syntax_support import *
+  from gyb_syntax_support.kinds import lowercase_first_word
+  from gyb_syntax_support.kinds import SYNTAX_BASE_KINDS
+  # -*- mode: Swift -*-
+  # Ignore the following admonition it applies to the resulting .swift file only
+
+  def syntax_buildable_child_type(syntax_kind, is_token=False, is_optional=False):
+    if syntax_kind in SYNTAX_BASE_KINDS:
+      buildable_type = syntax_kind + 'Buildable'
+    elif not is_token:
+      buildable_type = syntax_kind
+    elif 'List' in syntax_kind:
+      buildable_type = syntax_kind + 'Syntax'
+    else:
+      buildable_type = 'TokenSyntax'
+
+    if is_optional:
+      buildable_type += '?'
+
+    return buildable_type
+}%
+//// Automatically Generated From DeclBuildables.swift.gyb.
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+// MARK: Protocols
+
+% for kind in SYNTAX_BASE_KINDS:
+%   if kind != 'SyntaxCollection':
+%     build_kind = 'Syntax' if kind == 'Syntax' else kind + 'Syntax'
+%     if kind == 'Syntax':
+public protocol ${kind}ListBuildable {
```

Comment:
Should we add a convenience extension without `leadingTrivia` so we don't need to pass `nil`?